src/Microsoft.PowerShell.ConsoleHost/host/msh/UpdatesNotification.cs

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Management.Automation.Host;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.NetworkInformation;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// A Helper class for printing notification on PowerShell startup when there is a new update.
+    /// </summary>
+    internal static class UpdatesNotification
+    {
+        private const string UpdateCheckOptOutEnvVar = "POWERSHELL_UPDATECHECK_OPTOUT";
+        private const string Last3ReleasesUri = "https://api.github.com/repos/PowerShell/PowerShell/releases?per_page=3";
```

Comment:
A bit concerned about using GitHub directly:

- they may throttle if we have too many users checking at the same time (like in the start of business)
- they return a lot more info than we actually use

I was thinking we would have a read-only Azure blob that has the latest version information that gets updated whenever we publish new versions.

Reply:
@TravisEz13 created a static link `https://aka.ms/pwsh-buildinfo-daily` that points to a `json` file including the daily build tag. Currently there is no equivalent of this for `preview build`, `stable build` and `servicing build`, but he planed to add those too, so every release build will update the json file appropriately and we can query those json files for version information. I think we can use GitHub Rest API for now, so this can get in Preview.5. Once we have the equivalent static links for `preview` and `stable` builds, we can switch to those links. I think we can make this happen before RC.

Reply:
Ok, that seems fine then.